src/NUnitFramework/framework/Internal/ExceptionHelper.cs

```diff
@@ -33,6 +34,41 @@ namespace NUnit.Framework.Internal
     /// </summary>
     public class ExceptionHelper
     {
+#if !NET_4_5 && !PORTABLE
+        private static readonly Action<Exception> PreserveStackTrace;
+
+        static ExceptionHelper()
+        {
+            var method = typeof(Exception).GetMethod("InternalPreserveStackTrace", BindingFlags.Instance | BindingFlags.NonPublic);
+
+            if (method != null)
+            {
+                try
+                {
+                    PreserveStackTrace = (Action<Exception>)Delegate.CreateDelegate(typeof(Action<Exception>), method);
+                    return;
+                }
+                catch (InvalidOperationException) { }
+            }
+            PreserveStackTrace = _ => { };
+        }
+#endif
+
+        /// <summary>
+        /// Rethrows an exception, preserving its stack trace
+        /// </summary>
+        /// <param name="exception">The exception to rethrow</param>
+        public static void Rethrow(Exception exception)
```

Comment:
Making `ExceptionHelper` internal works fine, but it will probably break someone's code. Admittedly, they shouldn't be using something that is in a namespace named "Internal", but:
- as soon as something is made public, you can't really blame people for using it
- it might not even be intentional. Tools like ReSharper can automatically import missing namespaces. If the user referenced a class or an extension method with the same name as one from the `Internal` namespace and used this feature, R# could have "accidentally" imported the `Internal` namespace. It happened to me a few times (not sure if it was with NUnit or something else).